### ISSUE TITLE
Enhance Pytorch Estimator Callbacks interface with `logs` input

### ISSUE THREAD

**Issue by yushan111** [intel-analytics/BigDL#4586]
Orca Pytorch Estimator Callbacks refers to [tf.keras callback](https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/Callback).

In tf.keras callback, the logs dictionary that callback methods take as argument will contain keys for quantities relevant to the current batch or epoch. For example, 
* In `on_epoch_end`, `logs` is a dictionary containing metric results for this training epoch, and for the validation epoch if validation is performed. Validation result keys are prefixed with val_. For training epoch, the values of the Model's metrics are returned. Example : {'loss': 0.2, 'accuracy': 0.7}.
* In `on_train_batch_ends`, `logs` is a dictionary containing aggregated metric results up until this batch.

We should also enable users using the `logs` argument in Orca Pytorch Estimator Callbacks to retrieve the training metric results within their own customized callbacks. 

Tasks:

- [ ] `logs` in `on_epoch_end`
- [ ] `logs` in  `on_train_batch_ends`
